index.js

```diff
@@ -51,27 +44,38 @@ function uglifyify(file, opts) {
     )
 
     debug = opts.sourceMap !== false && (debug || matched)
-    opts  = extend({}, {
+
+    var thisopts  = extend({}, {
       sourceMap: {
         filename: file
       }
     }, opts)
 
-    if (typeof opts.compress === 'object') {
-      delete opts.compress._
+    // remove exts before passing opts to uglify
+    delete thisopts.global
+    delete thisopts.exts
+    delete thisopts.x
+    delete thisopts._
+    delete thisopts._flags
+
+    if (typeof thisopts.compress === 'object') {
+      delete thisopts.compress._
     }
 
-    if (debug) opts.sourceMap.url = 'out.js.map'
+    if (debug) thisopts.sourceMap.url = 'out.js.map'
 
     // Check if incoming source code already has source map comment.
     // If so, send it in to ujs.minify as the inSourceMap parameter
     if (debug && matched) {
-      opts.sourceMap.content = convert.fromJSON(
+      thisopts.sourceMap.content = convert.fromJSON(
         new Buffer(matched[1], 'base64').toString()
       ).sourcemap
     }
 
-    var min = ujs.minify(buffer, opts)
+    var min = ujs.minify(buffer, thisopts)
+
+    if (min.error)
```

Comment:
this is a stream, so we should be emitting an error rather than throwing it (i added this is in #71 when i remapped the argv stuff)